### PR TITLE
[MediaLibrary][Android] Fix media's width and height are sometimes 0

### LINKED ISSUE #4326: Bug in MediaLibrary, where the media's width and height are 0
### Environment

  Expo CLI 2.6.0 environment info:
    System:
      OS: macOS High Sierra 10.13.6
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.14.1 - /usr/local/bin/node
      Yarn: 1.16.0 - /usr/local/bin/yarn
      npm: 6.4.1 - /usr/local/bin/npm
    IDEs:
      Android Studio: 3.2 AI-181.5540.7.32.5056338
      Xcode: 10.1/10B61 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^32.0.0 => 32.0.6 
      react: 16.5.0 => 16.5.0 
      react-native: https://github.com/expo/react-native/archive/sdk-32.0.0.tar.gz => 0.57.1 
    npmGlobalPackages:
      expo-cli: 2.6.0

I am testing on an Android device.

### Steps to Reproduce

I am getting the assets from my media files via `MediaLibrary.getAssetsAsync(options)`. After it is done, I am displaying it in an Image component with respect to the image's width and height and screen width. Meaning the total width of the image will be screen's width, and the image's height will resize according to the image's aspect ratio.

### Expected Behavior

I was expecting to get the original Image's width and height. Because, if I check the image details on my phone, its width and height are not 0.

### Actual Behavior

But, there are some files that have width:0 and height: 0. Especially images, that were uploaded to the Instagram file, which are located on the Instagram file directory. However, there are exif data (ImageWidth and ImageLength), that actually match with the image's height and width.

I have tested this on two separate android devices, and they behave exactly the same.

Screenshot_2019-05-28-13-59-55

### Reproducible Demo

Here is the link to a snack where you can reproduce the bug. [MediaLibraryBug Snack](https://snack.expo.io/@kakarn/medialibrarybug).

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Resolve #4326.

# How

For some images (for example from Instagram) dimensions got from `query` don't have value, so i get them from `BitmapFactory.Options`

# Test Plan

You can check it on https://snack.expo.io/@kakarn/medialibrarybug.

### CODE REVIEW

File: packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryUtils.java [hunk 1]
```diff
@@ -215,6 +213,27 @@ static Integer convertMediaType(String mediaType) throws IllegalArgumentExceptio
     return MEDIA_TYPES.get(mediaType);
   }
 
+  static int[] getSizeFromCoursor(Cursor cursor, int mediaType, int localUriIndex){
```
Comment (tsapeta) [inline packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryUtils.java]:
```suggestion
  static int[] getSizeFromCursor(Cursor cursor, int mediaType, int localUriIndex) {
```

File: packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryUtils.java [hunk 2]
```diff
@@ -146,7 +144,7 @@ static void putAssetsInfo(Cursor cursor, ArrayList<Bundle> response, int limit,
     for (int i = 0; i < limit && !cursor.isAfterLast(); i++) {
       String localUri = "file://" + cursor.getString(localUriIndex);
       int mediaType = cursor.getInt(mediaTypeIndex);
-      int[] size = maybeRotateAssetSize(cursor.getInt(widthIndex), cursor.getInt(heightIndex), cursor.getInt(orientationIndex));
+      int[] size = getSizeFromCoursor(cursor, mediaType, localUriIndex);
```
Comment (tsapeta) [inline packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryUtils.java]:
```suggestion
      int[] size = getSizeFromCursor(cursor, mediaType, localUriIndex);
```

**Review by tsapeta — CHANGES_REQUESTED** [review]
Looks good except one typo 😉 
Would you like to add some tests to `test-suite` checking whether returned sizes are correct? I guess it is not possible to reproduce an issue in our tests, but simple checks on our test assets would be great and enough 🙂

**Review by tsapeta — APPROVED** [review]

File: packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryUtils.java [hunk 3]
```diff
@@ -215,6 +213,27 @@ static Integer convertMediaType(String mediaType) throws IllegalArgumentExceptio
...
 
+  static int[] getSizeFromCursor(Cursor cursor, int mediaType, int localUriIndex){
+    final int orientationIndex = cursor.getColumnIndex(Media.ORIENTATION);
+    final int widthIndex = cursor.getColumnIndex(Media.WIDTH);
+    final int heightIndex = cursor.getColumnIndex(Media.HEIGHT);
+
+    int[] size;
+    // If image doesn't have the required information, we can get them from Bitmap.Options
+    if (cursor.getType(widthIndex) == Cursor.FIELD_TYPE_NULL &&
+        cursor.getType(heightIndex) == Cursor.FIELD_TYPE_NULL &&
```
Comment (bbarthec) [inline packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryUtils.java]: I believe if any of `cursor.getType({widthIndex, heightIndex})` equals `Cursor.FIELD_TYPE_NULL` we should check for actual size using this different flow. 
```suggestion
if ((cursor.getType(widthIndex) == Cursor.FIELD_TYPE_NULL ||
    cursor.getType(heightIndex) == Cursor.FIELD_TYPE_NULL) &&
    mediaType == Files.FileColumns.MEDIA_TYPE_IMAGE)
```

**Review by bbarthec — APPROVED** [review]
LGTM 💯 Only one small suggestion.

File: apps/test-suite/tests/MediaLibrary.js [hunk 4]
```diff
@@ -205,6 +205,28 @@ export async function test(t) {
...
+        const options = { mediaType, album };
+        const { assets } = await MediaLibrary.getAssetsAsync(options);
+        t.expect(assets.length).toBe(IMG_NUMBER);
+        assets.forEach(asset => {
+          t.expect(asset.width).not.toEqual(0);
+          t.expect(asset.height).not.toEqual(0);
+        });      
+      });
+
+      t.it('check size - vide', async () => {
```
Comment (tsapeta) [inline apps/test-suite/tests/MediaLibrary.js]:
```suggestion
      t.it('check size - video', async () => {
```

**Review by tsapeta — APPROVED** [review]

**Pull request merged**